### PR TITLE
K8SSAND-973: Add support for service account annotations in helm chart

### LINKED ISSUE #1135: Unable to configure service account annotations for medusa-operator
## Bug Report

**Describe the bug**

Unable to configure service account annotations for medusa-operator. 

**To Reproduce**

Apply the following in your configuration:

```yaml
medusa-operator:
  serviceAccount:
    annotations:
      foo: "bar"
```

**Expected behavior**
The `k8ssandra-medusa-operator` service account should now have that annotation, however, it does not.

**Environment (please complete the following information):**
AWS EKS `v1.18.20`

* Helm charts version info
`1.3.1`

* Kubernetes version information:

`kubectl version`

* Kubernetes cluster kind:
AWS EKS

### PR DESCRIPTION
**Pull request by perplexa**
**What this PR does**:

Add support for service account annotations in helm chart

**Which issue(s) this PR fixes**:
Fixes #1135 

**Checklist**
- [x] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [ ] CHANGELOG.md updated (not required for documentation PRs)
- [x] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: charts/k8ssandra-common/Chart.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ name: k8ssandra-common
 description: |
   Helper library containing functions used by many of the K8ssandra stack Helm charts.
 type: library
-version: 0.28.4
+version: 0.28.5
```
Comment (jsanda) [inline charts/k8ssandra-common/Chart.yaml:6]: This is already 0.28.5 in main. Can you bump this to 0.28.6.

File: charts/k8ssandra/Chart.yaml [hunk 2]
```diff
@@ -18,7 +18,7 @@ dependencies:
     repository: file://../medusa-operator
     condition: medusa.enabled
   - name: k8ssandra-common
-    version: 0.28.4
+    version: 0.28.5
```
Comment (jsanda) [inline charts/k8ssandra/Chart.yaml:21]: All the other charts depend on k8ssandra-common as well. You will need to update the dependencies in other charts and then bump their chart versions.

**Review by jsanda — CHANGES_REQUESTED** [review]

**Comment by jsanda** [comment]
You need to rebase off main to fix the lint error.

**Comment by perplexa** [comment]
@jsanda done ✅

**Comment by perplexa** [comment]
@jsanda all done! 👍🏼

**Pull request merged**